authentication/auth.go

```diff
@@ -558,6 +556,32 @@ func GenerateTokenFromDefaultsV2(schemaURL string, accountServiceURL string, url
 	return token, ""
 }
 
+func addAdditionalMetadata(surveyMetadata map[string]interface{}, requiredSchemaMetadata []Metadata) []Metadata {
+	for _, metadataVal := range surveyMetadata {
+		for key, value := range metadataVal.(map[string]interface{}) {
+			if isExistingRequiredSchemaMetadata(requiredSchemaMetadata, key) {
+				continue
+			} else {
+				newMetadata := Metadata{}
```

Comment:
Don't think we need this line, can just use `newMetadata := Metadata{Name: key, Validator: "boolean", Default: "false"}` below?